exporter/splunkhecexporter/hec_worker.go

```diff
@@ -54,6 +58,12 @@ func (hec *defaultHecWorker) send(ctx context.Context, bufferState *bufferState,
 	}
 
 	if bufferState.compressionEnabled {
+		r, _ := gzip.NewReader(bytes.NewBuffer(bufferState.buf.Bytes()))
+		// Try to read 1 byte of uncompressed data. If EOF is raised then it means it's empty
+		// Splunk will return "No data" and reply_code=5 if it is empty.
+		if _, err1 := io.CopyN(io.Discard, r, 1); errors.Is(err1, io.EOF) {
+			return consumererror.NewPermanent(fmt.Errorf("Dropping this buffer as size of umcompressed data is 0 bytes"))
```

Comment:
No, this will be interpreted as the collector having failures. Can you add a length check upstream?

Reply:
I have moved the check inside `postEvents`.
The flow before:
- Check for buffer size and then call the `postEvents`
The flow after:
- Moved the check inside the `postEvents` and call `postEvents` directly.

The reason for this is, we can only apply the check on gzip after closing the writer. so `postEvents` would be ideal.